src/contract.rs

```diff
@@ -4061,6 +4073,79 @@ mod tests {
         assert_eq!(ask_storage.load("ask_id".as_bytes()).is_err(), true);
     }
 
+    #[test]
```

Comment:
only did a happy path sanity check test for these mostly copied from expired but with different action check